plugins/modules/consul_role.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022, Håkon Lerring
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: consul_role
+short_description: Manipulate Consul roles
+version_added: 7.2.0
+description:
+ - Allows the addition, modification and deletion of roles in a consul
+   cluster via the agent. For more details on using and configuring ACLs,
+   see https://www.consul.io/docs/guides/acl.html.
+author:
+  - Håkon Lerring (@Hakon)
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - A name used to identify the role.
+    required: true
+    type: str
+  state:
+    description:
+      - whether the role should be present or absent.
+    required: false
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  description:
+    description:
+      - Description of the role.
+    required: false
+    type: str
+    default: ''
+  policies:
+    type: list
+    elements: dict
+    description:
+      - List of policies to attach to the role.
+      - Each element must have a "name" or "id" (or both) to identify the policy. See consul_policy for more info.
+    required: false
+    default: []
```

Comment:
Does `default: []` mean that if the user does not specify `policies`, the module will remove all currently assigned policies?

Reply:
Correct, by not specifying any `policies` they will be unassigned from this role. The user can specify any or all of `policies`, `service_identities` or `node_identities` to be set (or unset) on a role.
I'll add this information to the documentation (e.g. "If not specified, any policies assigned to this role will be unassigned. ")

Reply:
How about not having a default, and if the user doesn't specify this value (i.e. you get `None` in the Python code) to not update it? Then users can distinguish between "do not update this value" and "clear all values" by not specifying it or specifying `[]` (empty list).

Reply:
Sure, this sounds like a good idea.